hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java

```diff
@@ -131,18 +146,18 @@ public RecordCacheImpl(
             @NonNull final WorkingStateAccessor workingStateAccessor,
             @NonNull final ConfigProvider configProvider) {
         this.deduplicationCache = requireNonNull(deduplicationCache);
+        this.workingStateAccessor = requireNonNull(workingStateAccessor);
         this.configProvider = requireNonNull(configProvider);
         this.histories = new ConcurrentHashMap<>();
 
-        rebuild(workingStateAccessor);
+        rebuild();
     }
 
     /**
-     * Rebuild the internal data structures based on the current working state. Called during startup and during
-     * reconnect. The amount of time it takes to rebuild this data structure is not dependent on the size of state, but
-     * rather, the number of transactions in the queue (which is capped by configuration at 3 minutes by default).
+     * The data structures can't be created from the state during restart or reconnect.
+     * They are just cleared during restart or reconnect.
      */
-    public void rebuild(final WorkingStateAccessor workingStateAccessor) {
+    public void rebuild() {
```

Comment:
With the change to the queue being TransactionReceiptEntry (previously TransactionRecordEntry) when rebuilding the queue the asTxnRecord only build TransactionRecord's with the receipt and transactionID. I'm just thinking that we may need to do something to include the other information in the TransactionRecord otherwise it's lost and CryptoGetAccountRecordsQuery and TransactionGetRecordQuery will return mostly empty TransactionRecords. Thinking about this more.

Reply:
As discussed, we will not rely on queries returning correct values as per the issue